examples/gno.land/r/demo/foo721/foo721_test.gno

```diff
@@ -0,0 +1,31 @@
+package foo721
+
+import (
+	"testing"
+
+	"gno.land/p/demo/grc/grc721"
+	"gno.land/r/demo/users"
+)
+
+func TestFoo721(t *testing.T) {
+	admin := users.AddressOrName("g1us8428u2a5satrlxzagqqa5m6vmuze025anjlj")
+	hariom := users.AddressOrName("g1var589z07ppjsjd24ukm4uguzwdt0tw7g47cgm")
+
+	for i, tc := range []struct {
+		name     string
+		expected interface{}
+		fn       func() interface{}
+	}{
+		{"BalanceOf(admin)", 10, func() interface{} { return BalanceOf(admin) }},
+		{"BalanceOf(hariom)", 5, func() interface{} { return BalanceOf(hariom) }},
+		{"OwnerOf(0)", admin.Resolve(), func() interface{} { return OwnerOf(grc721.TokenID("0")) }},
+		{"IsApprovedForAll(admin, hariom)", false, func() interface{} { return IsApprovedForAll(admin, hariom) }},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			got := tc.fn()
+			if tc.expected != got {
+				t.Errorf("expected: %v got: %v", tc.expected, got)
+			}
+		})
+	}
```

Comment:
I find this test table really hard to read. 

This test should also be broken into multiple unit tests for easier management and cleaner code - there should be a unit test for each functionality that the contract provides.

Bundling functionality checks like this with `interface {}` is hard to manage and change.

I also suggest defining test tables as:

```go
// TestMyTest verifies ....
func TestMyTest(t *testing.T) {
	testTable := []struct {
		name string
		
		// ...
	} {
		{
			// ...
		},
	}
	
	for _, testCase := range testTable {
		t.Run(testCase.name, func(t *testing.T) {
			// ...
		})
	}
}
```

I'm not sure if we support `t.Parallel()` inside `gno` test files, but if we do, you can also use it here

Reply:
I already wrote extensive tests for [`/p/demo/grc/grc721`](https://github.com/harry-hov/gno/blob/hariom/grc721/examples/gno.land/p/demo/grc/grc721/basic_nft_test.gno). 

And about readability, I believe it's subjective.

Reply:
Resolving this convo because you've already added unit tests for the package implementation